TRRandomizerCore/Levels/TR3CombinedLevel.cs

```diff
@@ -114,5 +119,16 @@ public TR3Adventure Adventure
                 return TR3Adventure.India;
             }
         }
+
+        public void RemoveModel(TR3Entities type)
+        {
+            List<TRModel> models = Data.Models.ToList();
+            if (models.Find(m => m.ID == (uint)type) is TRModel model)
+            {
+                models.Remove(model);
+                Data.Models = models.ToArray();
+                Data.NumModels--;
```

Comment:
Outside of the scope of this PR but `NumModels` looks like a bug waiting to happen if you ever forget to update it - it should just be able to take the length of the array. It could be 

You can keep the property for getting:
```
public int NumModels => Models.Length;
```

Reply:
Agreed. I'd like to change all the arrays in the level classes to lists to avoid this kind of thing, which we have in a *lot* of places.
```
List<T> list = level.ThingArray.ToList();
list.Add(new T());
level.ThingArray = list.ToArray();
level.NumThings++;
```
The goal being `level.ThingList.Add(new T());` and nothing else.

It'll be a sizeable change, but less code in the end. I'll raise an issue to tackle this separately.